### DISCUSSION TITLE
`pasta unknown version` in `podman info` on arch linux

### DISCUSSION THREAD

**Discussion by MartinX3** [containers/podman#22148] [Q&A] [answered]
I don't know if it's a bug, but `podman info` shows me on arch linux
```
pasta:
    executable: /usr/bin/pasta
    package: /usr/bin/pasta ist in passt 2024_03_20.71dd405-1 enthalten
    version: |
      pasta unknown version
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
```

and I don't know if `pasta unknown version` is normal.

**Comment by Luap99** [comment]
That is should be the output of `pasta --version`, it works fine on fedora so most likely a package bug, I suggest you file this as bug against your distro package.

**Reply by MartinX3** [reply]
Thank you, I created an issue ticket at the arch linux package.

**Comment by afbjorklund** [comment, marked as answer]
> and I don't know if `pasta unknown version` is normal.

It is normal, unless you use git instead of tarballs for building:

```Makefile
VERSION ?= $(shell git describe --tags HEAD 2>/dev/null || echo "unknown\ version")
```

So your packaging build script need to pass VERSION explicitly:

https://src.fedoraproject.org/rpms/passt/blob/rawhide/f/passt.spec

`%make_build passt VERSION="%{version}-%{release}.%{_arch}"`

**Reply by MartinX3** [reply]
Thank you, I created an issue ticket at the arch linux package.

**Reply by afbjorklund** [reply]
I think the release and arch is specific to fedora, only needs version.

`make VERSION=2024_03_20.71dd405`

**Reply by afbjorklund** [reply]
https://gitlab.archlinux.org/archlinux/packaging/packages/passt/-/issues/1

**Reply by afbjorklund** [reply]
It could have been worse, for Buildroot it reports the OS version instead...

**Comment by afbjorklund** [comment]
`    package: /usr/bin/pasta ist in passt 2024_03_20.71dd405-1 enthalten`

This is a bug in [containers-common](https://github.com/containers/common), there is no filtering for Arch (yet)

* https://github.com/containers/common/issues/1934

Like there is for Alpine:

```go
                if cmdArg[0] == "/sbin/apk" {
                        prefix := cmdArg[len(cmdArg)-1] + " is owned by "
                        output = strings.Replace(output, prefix, "", 1)
                }

```


It probably also need to set the locale to `C`, to avoid the german...
* https://github.com/containers/common/issues/1933

Also needs to filter out the `Copyright` text, after the version output.
* https://github.com/containers/common/issues/1931

**Reply by MartinX3** [reply]
Dankeschön, I created an issue ticket at the arch linux package.

**Reply by afbjorklund** [reply]
The redundant git information is actually part of the upstream tagging...

```console
$ git describe --tags --long 2024_03_20.71dd405
2024_03_20.71dd405-0-g71dd405
```

https://passt.top/passt/refs/tags

It doesn't use version numbers, but dates and commits.
